drivers/ethernet/dwc_xgmac/eth_dwc_xgmac.c

```diff
@@ -0,0 +1,1973 @@
+	reg_val = sys_read32(reg_addr);
+#ifdef CONFIG_ETH_DWC_XGMAC_BOTTOM_HALF_WORK_QUEUE
+	cntxt_data->mac_interrupt_sts |= reg_val;
+#else
+	eth_dwc_xgmac_mac_isr(dev, reg_val);
+#endif
+
+	/* submitting work item for interrupt bottom half processing.
+	 * Checking if already a work item is waiting in the work queue to process the interrupt
+	 * flags
+	 */
+#ifdef CONFIG_ETH_DWC_XGMAC_BOTTOM_HALF_WORK_QUEUE
+	k_work_submit(&data->isr_work);
+#endif
+}
+/**
+ * @brief This is the expiry function which will be called from the
+ * system timer irq handler upon configured isr polling timer expires.
+ *
+ * @param timer Pointer to the timer object
+ */
+#ifdef CONFIG_ETH_DWC_XGMAC_POLLING_MODE
+static void eth_dwc_xgmac_irq_poll(struct k_timer *timer)
+{
+	struct eth_dwc_xgmac_dev_data *dev_data = (struct eth_dwc_xgmac_dev_data *)CONTAINER_OF(
+		timer, struct eth_dwc_xgmac_dev_data, isr_polling_timer);
+	const struct device *dev = dev_data->irq_cntxt_data.dev;
+
+	eth_dwc_xgmac_isr(dev);
+}
+#endif /* CONFIG_ETH_DWC_XGMAC_POLLING_MODE */
+
+/**
+ * @brief XGMAC device initialization function
+ * Initializes the XGMAC itself, the DMA memory area used by the XGMAC.
+ *
+ * @param dev Pointer to the ethernet device
+ * @retval 0 if the device initialization completed successfully
+ *         -ENODEV if reset is applicable and if reset device is not available or not ready.
+ *         -ENOSYS if reset is applicable and if it failes to toggle the reset signal.
+ *         -ETIMEDOUT if XGMAC software reset gets timeout.
+ */
+static int eth_dwc_xgmac_dev_init(const struct device *dev)
+{
+	const struct eth_dwc_xgmac_config *const config =
+		(struct eth_dwc_xgmac_config *)dev->config;
+	struct eth_dwc_xgmac_dev_data *const data = (struct eth_dwc_xgmac_dev_data *)dev->data;
+	mem_addr_t ioaddr;
+	int ret = 0;
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(resets)
+
+	if (config->reset.dev != NULL) {
+		if (!device_is_ready(config->reset.dev)) {
+			LOG_ERR("%s, Reset device is not ready", dev->name);
+			return -ENODEV;
+		}
+		ret = reset_line_toggle(config->reset.dev, config->reset.id);
+		if (ret) {
+			LOG_ERR("%s: Failed to reset peripheral", dev->name);
+			return ret;
+		}
+	} else {
+		LOG_ERR("%s, Reset device is not available", dev->name);
+		return -ENODEV;
+	}
+
+#endif
+
+	DEVICE_MMIO_MAP(dev, K_MEM_CACHE_NONE);
+	ioaddr = get_reg_base_addr(dev);
+
+	/*
+	 * Initialization procedure as described in the dwc xgmac 10G Ethernet MAC data book.
+	 */
+	ret = dwxgmac_software_reset(ioaddr);
+	if (ret) {
+		LOG_ERR("%s: MAC reset timeout", dev->name);
+		return ret;
+	}
+
+	dwxgmac_dma_init(dev, &config->dma_cfg);
+
+	dwxgmac_dma_desc_init(config, data);
+
+	dwxgmac_dma_chnl_init(dev, config, data);
+
+	dwxgmac_dma_mtl_init(dev, config);
+
+	dwxgmac_mac_init(dev, config, data);
+
+	/* set MAC address */
+	dwxgmac_set_mac_addr_by_idx(dev, data->mac_addr, 0, false);
+
+	dwxgmac_irq_init(dev);
+	LOG_INF("XGMAC ethernet driver init done");
```

Comment:
I suggest to avoid using info level as normally they are not that useful. So I would just do `LOG_DBG()` here (same comment to all the info messages in this PR)

Reply:
Replaced LOG_INF with LOG_DBG